ha/core/event_analyzer/parser/parser.py

```diff
@@ -159,15 +159,102 @@ class ClusterResourceParser(Parser):
+        self.predefined_config = {"cluster_id": self.cluster_id, "site_id": self.site_id, "rack_id": self.rack_id}
+        self.system_health_node_key_template = "cortx/ha/system/cluster/$cluster_id/site/$site_id/rack/$rack_id/node/$node_id/health"
         Log.info("ClusterResource Parser is initialized ...")
 
-    def parse_event(self, msg: str) -> HealthEvent:
+    def prepare_key(self, res_alert: dict) -> str:
+        """Prepares a key to be searched in confstore"""
+        replacable_patterns = re.findall(r"\$\w+", self.system_health_node_key_template)
+        for pattern in replacable_patterns:
+            replacable_string = pattern[1:]
+            if replacable_string in self.predefined_config:
+                config_val = self.predefined_config[replacable_string]
+                key = self.system_health_node_key_template.replace(pattern, config_val)
+            else:
+                config_val = res_alert["_resource_name"]
+                key = self.system_health_node_key_template.replace(pattern, config_val)
+            self.system_health_node_key_template = key
+        return self.system_health_node_key_template
+
+    def _create_health_event_format(self, cluster_resource_alert: dict, specific_info: dict = None) -> dict:
+        """
+        Creates the defined structure to send an helath event to system health
+        for further processing
+        """
+        timestamp = str(int(time.time()))
+        event_id = timestamp + str(uuid.uuid4().hex)
+        node_id = cluster_resource_alert["_resource_name"]
+        resource_type = cluster_resource_alert["_resource_type"]
+        event_type = cluster_resource_alert["_event_type"]
+        timestamp = cluster_resource_alert["_timestamp"]
+
+        event = {
+                    EVENT_ATTRIBUTES.EVENT_ID : event_id,
+                    EVENT_ATTRIBUTES.EVENT_TYPE : event_type,
+                    EVENT_ATTRIBUTES.SEVERITY : StatusMapper.EVENT_TO_SEVERITY_MAPPING[event_type],
+                    EVENT_ATTRIBUTES.SITE_ID : self.site_id, # TODO: Should be fetched from confstore
+                    EVENT_ATTRIBUTES.RACK_ID : self.rack_id, # TODO: Should be fetched from confstore
+                    EVENT_ATTRIBUTES.CLUSTER_ID : self.cluster_id, # TODO: Should be fetched from confstore
+                    EVENT_ATTRIBUTES.STORAGESET_ID : node_id,
+                    EVENT_ATTRIBUTES.NODE_ID : node_id,
+                    EVENT_ATTRIBUTES.HOST_ID : node_id,
+                    EVENT_ATTRIBUTES.RESOURCE_TYPE : resource_type,
+                    EVENT_ATTRIBUTES.TIMESTAMP : timestamp,
+                    EVENT_ATTRIBUTES.RESOURCE_ID : node_id,
+                    EVENT_ATTRIBUTES.SPECIFIC_INFO : specific_info
+                }
+
+        Log.debug(f"Parsed {event} schema")
+        # Log.error(f"########## Parsed {event} schema")
+        return event
+
+    @staticmethod
+    def _get_sys_health_key_val(key_to_search: str, conf_store) -> (dict,int,str):
```

Comment:
Shouldn't all system health related logic be in the system health module ?

Reply:
Yes. Moved.